components/places/src/db/schema.rs

```diff
@@ -111,6 +111,48 @@ const CREATE_TRIGGER_AFTER_INSERT_ON_PLACES: &str = "
     END
 ";
 
+// Triggers which update visit_count and last_visit_date based on historyvisits
+// table changes.
+const  EXCLUDED_VISIT_TYPES: &str = "0, 4, 7, 8, 9"; // stolen from desktop
+
+lazy_static! {
+    static ref CREATE_TRIGGER_HISTORYVISITS_AFTERINSERT: String = format!("
+        CREATE TEMP TRIGGER moz_historyvisits_afterinsert_trigger
+        AFTER INSERT ON moz_historyvisits FOR EACH ROW
+        BEGIN
+            UPDATE moz_places SET
+                visit_count_remote = visit_count_remote + (SELECT NEW.visit_type NOT IN ({}) AND NOT(NEW.is_local)),
+                visit_count_local =  visit_count_local + (SELECT NEW.visit_type NOT IN ({}) AND NEW.is_local),
+                last_visit_date_local = MAX(IFNULL(last_visit_date_local, 0),
```

Comment:
This will turn last_visit_date_local from NULL to 0 if it was NULL (In Desktop places, this isn't an issue since there's only one last_visit_date, and it's never null after a visit is inserted). Ditto for remote slightly below.

After some discussion in IRC, I think it's worth considering making them NOT NULL DEFAULT 0 or something, since it seems less error prone (and dealing with Option<T> is a bit of a pain in rusqlite...).

@linacambridge (who has more experience with SQL and places than me) might have better insight here as to what is sensible though.

Reply:
done!